sklearn/neighbors/quad_tree.pxd

```diff
@@ -0,0 +1,91 @@
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: cdivision=True
+# Author: Thomas Moreau <thomas.moreau.2010@gmail.com>
+# Author: Olivier Grisel <olivier.grisel@ensta.fr>
+
+# See quad_tree.pyx for details.
+
+import numpy as np
+cimport numpy as np
+
+ctypedef np.npy_float32 DTYPE_t          # Type of X
+ctypedef np.npy_intp SIZE_t              # Type for indices and counters
+ctypedef np.npy_int32 INT32_t            # Signed 32 bit integer
+ctypedef np.npy_uint32 UINT32_t          # Unsigned 32 bit integer
+
+# This is effectively an ifdef statement in Cython
+# It allows us to write printf debugging lines
+# and remove them at compile time
+cdef enum:
+    DEBUGFLAG = 1
+
+cdef float EPSILON = 1e-6
+
+# XXX: Careful to not change the order of the arguments. It is important to
+# have is_leaf and max_width consecutive as it permits to avoid padding by
+# the compiler and keep the size coherent for both C and numpy data structures.
```

Comment:
This is the case in this code. This comment was added on purpose after several failures to get a working conversion for `_QuadTree.cells` to `ndarray`.
It seems to be working for the compilers used on travis and appveyor (see the check [here](https://github.com/tomMoral/scikit-learn/blob/optim_tsne/sklearn/neighbors/quad_tree.pyx#L60) which ensures that the `cython` object and the created `numpy` data structure have the same size)